nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/providers/OAuth2ServiceProviderRegistry.java

```diff
@@ -27,6 +27,24 @@
 
     OAuth2ServiceProvider getProvider(String serviceName);
 
-    OAuth2ServiceProvider addProvider(String serviceName, String description, String tokenServerURL, String authorizationServerURL,
-            String clientId, String clientSecret, List<String> scopes);
+    List<OAuth2ServiceProvider> getProviders();
+
+    OAuth2ServiceProvider addProvider(String serviceName, String description, String tokenServerURL,
+        String authorizationServerURL, String clientId, String clientSecret,
+        List<String> scopes);
+
+    OAuth2ServiceProvider addProvider(String serviceName, String description, String tokenServerURL,
+        String authorizationServerURL, String userAuthorizationURL, String clientId, String clientSecret,
+        List<String> scopes, Boolean isEnabled);
```

Comment:
This is getting unwieldy. We should really use a builder pattern in all of our APIs that have lots of arguments. For a later refactoring. Is the original method still used? Can't it be deprecated right now?